### PR TITLE
Add 'daily and sports activities' from UCIrvine ML Repository

### PR DESCRIPTION
**Pull request by VincentRouvreau**
and the script to filter data and generate left_leg_magnetometer.csv
cf. https://archive-beta.ics.uci.edu/ml/datasets/daily+and+sports+activities

git lfs has been activated for this dataset as the pandas pickle file is about 111Mo. To be discussed in code review.

_Please use the following template to help us managing data copyrights._

## In case of a new dataset

Checklist
- [X] My dataset is in its own directory (one per directory)
- [X] This directory also contains a proper `LICENSE` file
- [X] The dataset is generated by the author of this PR
- [X] The dataset is taken from the web, the URL is: _Please provide the url so we can check the license_

### CODE REVIEW

**Comment by mglisse** [comment]
> git lfs has been activated for this dataset as the pandas pickle file is about 111Mo. To be discussed in code review.

If I understand, the main difference of git lfs is that by default a clone will not download the history, only the files that are actually in the current branch. Sounds good, although we are unlikely to have many file changes or removals in this repository. Also, github only accepts big files if they use git lfs. One limitation is that I don't think we can exceed 1G of total git lfs storage?

File: points/activities/activities.LICENSE [hunk 1]
```diff
@@ -0,0 +1,2 @@
+The 'daily and sports activities' dataset comes from the UC Irvine Machine Learning Repository https://archive-beta.ics.uci.edu/ml/datasets/daily+and+sports+activities
+Please read carefully the terms of use of this dataset (in the above link) before using it.
```
Comment (mglisse) [inline points/activities/activities.LICENSE:2]: Does it say more than "CC BY 4.0"?

File: points/activities/data_selecta.py [hunk 2]
```diff
@@ -0,0 +1,53 @@
+import os
+import pandas as pd
+
+#
+# This program requires to launch data from https://archive-beta.ics.uci.edu/ml/datasets/daily+and+sports+activities
```
Comment (mglisse) [inline points/activities/data_selecta.py:5]: s/launch/load/ ? :rocket: 
(+ broken URL)

File: points/activities/data_selecta.py [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+# Unzip the data and copy data_selecta.py in a same directory data folder
+# Launch 'python data_selecta.py'
+#
+
+# Will be auto-saved as a xz compressed pickle file
+result_file = 'activities.xz'
```
Comment (mglisse) [inline points/activities/data_selecta.py:11]: Did you compare the size with `activities.zst`?

Comment (VincentRouvreau) [reply points/activities/data_selecta.py:11]: My _"benchmark"_:
`to_pickle('activities.xz')` : 111 Mo
`to_pickle('activities.bz2')` : 140 Mo
`to_pickle('activities.zst')` : 193 Mo
`to_pickle('activities.zip')` : 199 Mo
`to_pickle('activities.gz')` : 199 Mo
`to_pickle('activities.pkl')` : 410 Mo

`to_csv('activities.csv.xz')` : 144 Mo
`to_csv('activities.csv.gz')` : 160 Mo
`to_csv('activities.csv.zst')` : 170 Mo
`to_csv('activities.csv')` : 418 Mo

**Comment by VincentRouvreau** [comment]
> > git lfs has been activated for this dataset as the pandas pickle file is about 111Mo. To be discussed in code review.
> 
> If I understand, the main difference of git lfs is that by default a clone will not download the history, only the files that are actually in the current branch. Sounds good, although we are unlikely to have many file changes or removals in this repository. Also, github only accepts big files if they use git lfs. One limitation is that I don't think we can exceed 1G of total git lfs storage?

In [github lfs documentation](https://docs.github.com/en/repositories/working-with-files/managing-large-files/about-git-large-file-storage), they are talking of a limit at 2GB for a file size.
An yes, you have to install `git lfs` and then specify which type of file you want git lfs to track (`git lfs track "*.xz"` in our case)

File: points/activities/data_selecta.py (hunk 3 above)
Comment (mglisse) [inline points/activities/data_selecta.py:11]: I don't know if there is a convention, whether the name should be activities.xz, activities.pkl (with an explicit compression parameter), or activities.pkl.xz. A quick search found all three, so I guess your name is fine.

Comment (VincentRouvreau) [reply points/activities/data_selecta.py:11]: I don't know... I explained it in the README.md, but maybe `.pkl.xz` is more explicit. Tell me your preference

File: points/activities/data_selecta.py (hunk 2 above)
Comment (VincentRouvreau) [reply points/activities/data_selecta.py]: Done on [36b0782](https://github.com/GUDHI/gudhi-data/pull/7/commits/36b0782396de99795894102ce0410245498115c5). Just hope the url won't change to often...

File: points/activities/activities.LICENSE (hunk 1 above)
Comment (VincentRouvreau) [reply points/activities/activities.LICENSE:2]: No, but if license change in the future, I prefer to provide a link in order to have less maintenance.

Comment (VincentRouvreau) [reply points/activities/activities.LICENSE]: You are right. I added CC BY 4.0 in the README.md and in activities.LICENSE on [36b0782](https://github.com/GUDHI/gudhi-data/pull/7/commits/36b0782396de99795894102ce0410245498115c5)

**Comment by mglisse** [comment]
I ran `pandas.read_pickle('activities.xz')` with pandas-1.3.5 and got
```
---------------------------------------------------------------------------
AttributeError                            Traceback (most recent call last)
Cell In [2], line 1
----> 1 pandas.read_pickle('activities.xz')

File /usr/lib/python3/dist-packages/pandas/io/pickle.py:222, in read_pickle(filepath_or_buffer, compression, storage_options)
    217             return pickle.load(handles.handle)  # type: ignore[arg-type]
    218     except excs_to_catch:
    219         # e.g.
    220         #  "No module named 'pandas.core.sparse.series'"
    221         #  "Can't get attribute '__nat_unpickle' on <module 'pandas._libs.tslib"
--> 222         return pc.load(handles.handle, encoding=None)
    223 except UnicodeDecodeError:
    224     # e.g. can occur for files written in py27; see GH#28645 and GH#31988
    225     return pc.load(handles.handle, encoding="latin-1")

File /usr/lib/python3/dist-packages/pandas/compat/pickle_compat.py:274, in load(fh, encoding, is_verbose)
    271         up = Unpickler(fh)
    272     up.is_verbose = is_verbose
--> 274     return up.load()
    275 except (ValueError, TypeError):
    276     raise

File /usr/lib/python3.10/pickle.py:1213, in _Unpickler.load(self)
   1211             raise EOFError
   1212         assert isinstance(key, bytes_types)
-> 1213         dispatch[key[0]](self)
   1214 except _Stop as stopinst:
   1215     return stopinst.value

File /usr/lib/python3.10/pickle.py:1538, in _Unpickler.load_stack_global(self)
   1536 if type(name) is not str or type(module) is not str:
   1537     raise UnpicklingError("STACK_GLOBAL requires str")
-> 1538 self.append(self.find_class(module, name))

File /usr/lib/python3/dist-packages/pandas/compat/pickle_compat.py:206, in Unpickler.find_class(self, module, name)
    204 key = (module, name)
    205 module, name = _class_locations_map.get(key, key)
--> 206 return super().find_class(module, name)

File /usr/lib/python3.10/pickle.py:1582, in _Unpickler.find_class(self, module, name)
   1580 __import__(module, level=0)
   1581 if self.proto >= 4:
-> 1582     return _getattribute(sys.modules[module], name)[0]
   1583 else:
   1584     return getattr(sys.modules[module], name)

File /usr/lib/python3.10/pickle.py:331, in _getattribute(obj, name)
    329         obj = getattr(obj, subpath)
    330     except AttributeError:
--> 331         raise AttributeError("Can't get attribute {!r} on {!r}"
    332                              .format(name, obj)) from None
    333 return obj, parent

AttributeError: Can't get attribute '_unpickle_block' on <module 'pandas._libs.internals' from '/usr/lib/python3/dist-packages/pandas/_libs/internals.cpython-310-x86_64-linux-gnu.so'>
```
It seems to work with pandas-1.5.1, so there are some requirements to be able to use this file. I don't know if there are some options to build the file in a way that is compatible with more versions of pandas.

**Comment by VincentRouvreau** [comment]
> .read_pickle('activities.xz')

I was able to reproduce the issue with pandas 1.3.5. I rebuilt the data with pandas 1.1.0 and now I am able to read the pickled data with pandas 1.1.0, 1.3.5 and 1.5.2. If you could try again, please ?

File: points/activities/data_selecta.py (hunk 3 above)
Comment (VincentRouvreau) [reply points/activities/data_selecta.py:13]: Pandas pickle file type seems to be a bad idea: https://docs.python.org/3/library/pickle.html
`to_csv('activities.csv.xz')` would be a better candidate, but not sure anymore that github LFS is a good place to store such data. Let me have a look to figshare (sklearn seems to us e this place), zenodo or even recherche.data.gouv.fr

Comment (VincentRouvreau) [reply points/activities/data_selecta.py:13]: Let's go with `to_csv('activities.csv.xz')`. The link to upload the git LFS file is a bit *hackish* and may not be robust, but let's try this solution. @mglisse do you agree ?

**Pull request merged**